### PR TITLE
feat: add resumable as a prop

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Added the ability for uploads to be resumable or not. In doing so had to write a hack to fix the Storage lib's code. 

Need some help on my Typescript. 

#### Description of how you validated changes

https://user-images.githubusercontent.com/321279/200904115-714aaf32-631a-4cf8-ad37-e909a9a960d2.mp4

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ErikCH — APPROVED** [review]
Looks good, other then what we chatted about offline about the file uploader pause bug. Which we can do in this PR or next.

**Review by dbanksdesign — COMMENTED** [review]
Need some TS help on a function returning different types based on if a prop is true or false.

File: packages/ui/src/helpers/storage/fileUploader/utils/uploader.ts [hunk 1]
```diff
@@ -19,21 +19,37 @@ export function uploadFile({
...
+  if (resumable === true) {
+    return Storage.put(fileName, file, {
+      level,
+      resumable,
+      progressCallback,
+      errorCallback,
+      // hack to remove the bucket name at the beginning
+      completeCallback: (event) => {
+        completeCallback({ key: event.key.split('/').slice(1).join('/') });
+      },
```
[the remark this answers is not in the document]
Comment (dbanksdesign) [reply packages/ui/src/helpers/storage/fileUploader/utils/uploader.ts:43]: Added a comment to remove this hack once v5 is released and we pull it in. The JS team fixed this behavior: https://github.com/aws-amplify/amplify-js/pull/10651

File: packages/react/src/components/Storage/FileUploader/Tracker/FileState.tsx [hunk 2]
```diff
@@ -12,12 +12,15 @@ export const FileState = ({
   percentage,
 }: FileStateProps): JSX.Element => {
   switch (fileState) {
-    case 'loading':
+    case 'loading': {
+      let text = translate('Uploading');
+      if (percentage > 0) text = text + `: ${percentage}%`;
```
Comment (0618) [inline packages/react/src/components/Storage/FileUploader/Tracker/FileState.tsx:17]: nit: 

```suggestion
      if (percentage > 0) text += `: ${percentage}%`;
```

File: packages/react/src/components/Storage/FileUploader/FileUploader.tsx [hunk 3]
```diff
@@ -9,6 +9,9 @@ import { Previewer } from './Previewer';
 import { UploadDropZone } from './UploadDropZone';
 import { Tracker } from './Tracker';
 
+const isUploadTask = (value: unknown): value is UploadTask =>
+  typeof (value as UploadTask)?.resume === 'function';
```
Comment (0618) [inline packages/react/src/components/Storage/FileUploader/FileUploader.tsx:13]: If we have to do `value as UploadTask`, why not define it at the beginning? 

```suggestion
const isUploadTask = (value: UploadTask): value is UploadTask =>
  typeof (value)?.resume === 'function';
```

**Review by 0618 — APPROVED** [review]
LGTM. Left non-blocker comments

**Review by wlee221 — APPROVED** [review]
LGTM

File: packages/react/src/components/Storage/FileUploader/FileUploader.tsx (hunk 3 above)
Comment (wlee221) [reply packages/react/src/components/Storage/FileUploader/FileUploader.tsx:13]: I believe it's a standard practice to have `value` defined loosely in typeguards: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards.

This typeguard, for example, asserts whether this loosely typed `value` is actually of type `UploadTask`, and narrows the type to it if so.

**Pull request merged**